packages/kit/src/runtime/server/index.js

```diff
@@ -144,15 +146,9 @@ export async function respond(request, options, state) {
 				const value = new_headers[key];
 
 				if (lower === 'set-cookie') {
-					const new_cookies = /** @type {string[]} */ (Array.isArray(value) ? value : [value]);
-
-					for (const cookie of new_cookies) {
-						if (cookies.includes(cookie)) {
-							throw new Error(`"${key}" header already has cookie with same value`);
-						}
-
-						cookies.push(cookie);
-					}
+					throw new Error(
+						`Use \`event.cookie.set(name, value, options)\` instead of \`event.setHeaders\` to set cookies`
+					);
```

Comment:
Instead of throwing here, couldn't we invoke something like `cookies.set(cookie.parse(value))` here?

Reply:
We _could_, but no-one wants to write this...

```js
import * as cookie from 'cookie';

export const actions = {
  default: async ({ setHeaders }) => {
    // ...
    setHeaders({
      'set-cookie': cookie.serialize('answer', 42, {
        httpOnly: true
      })
    });
   }
};
```

...when they could write this:

```js
export const actions = {
  default: async ({ cookies }) => {
    // ...
    cookies.set('answer', 42);
   }
};
```

I for one would be grateful for an error message that said 'you should delete a bunch of code, and you can probably remove the `cookie` dependency'.

It also feels very weird if you use `setHeaders` followed by `cookies.get` — it doesn't feel like it should work, and yet it does. Always better to have one way to do things.